narwhals/_pandas_like/dataframe.py

```diff
@@ -295,27 +296,9 @@ def join(
                 and (pd := get_pandas()) is not None
                 and parse_version(pd.__version__) < parse_version("1.4.0")
             ):
-
-                def generate_unique_token(
-                    n_bytes: int, columns: list[str]
-                ) -> str:  # pragma: no cover
-                    import secrets
-
-                    counter = 0
-                    while True:
-                        token = secrets.token_hex(n_bytes)
-                        if token not in columns:
-                            return token
-
-                        counter += 1
-                        if counter > 100:  # pragma: no cover
-                            msg = (
-                                "Internal Error: Narwhals was not able to generate a column name to perform cross "
-                                "join operation"
-                            )
-                            raise AssertionError(msg)
-
-                key_token = generate_unique_token(8, self.columns)
+                key_token = generate_unique_token(
+                    n_bytes=8, columns=[*self.columns, *other.columns]
+                )  # pragma: no cover
```

Comment:
do we need this? doesn't the `._implemention = 'modin'` in one of the tests which run by default hit this?